gr-iio/lib/device_sink_impl.cc

```diff
@@ -165,12 +167,38 @@ void device_sink_impl::channel_write(const struct iio_channel* chn,
         iio_channel_convert_inverse(chn, (void*)dst_ptr, (const void*)src_ptr);
 }
 
+void device_sink_impl::set_len_tag_key(const std::string& len_tag_key)
+{
+    if (!len_tag_key.size())
+        d_len_tag_key = pmt::PMT_NIL;
+    else
+        d_len_tag_key = pmt::string_to_symbol(len_tag_key);
+}
+
 int device_sink_impl::work(int noutput_items,
                            gr_vector_const_void_star& input_items,
                            gr_vector_void_star& output_items)
 {
     int ret;
 
+    if (d_len_tag_key != pmt::PMT_NIL) {
+        for (size_t i = 0; i < input_items.size(); i++) {
+            auto items_read = nitems_read(i);
+            get_tags_in_range(d_tags, i, items_read, items_read + 1, d_len_tag_key);
+
+            if (d_tags.size() != 1)
```

Comment:
When you have small IIO buffers I think this may break since the gr buffers could be larger. Meaning you could get multiple tagged buffers passed.